patch_recurrence/steps.py

```diff
@@ -50,8 +61,12 @@ def smoothen(scaled_imgs, patches, constants):
         img = scaled_imgs[k]
         patch = patches[k]
 
-        length_sd_array = img.shape[0] - patch_size
-        width_sd_array = img.shape[1] - patch_size
+        # For half patches
+        length_sd_array = width_sd_array = 0
+        for i in range(0, img.shape[0] - patch_size, 2):
+            length_sd_array += 1
+        for i in range(0, img.shape[1] - patch_size, 2):
```

Comment:
I thought so too about the loop, but no, it doesn't work directly. It depends on whether the img.shape[0] and img.shape[1] are even or odd. It changes for every image. Let me know if you can think of a better way.

Reply:
I can rewrite it by removing the for loop and putting a condition for odd even stuff

Reply:
No it doesn't, just use integer division (3 / 2 = 1, 2 / 2 = 1)

Reply:
The reason I didn't include step in constants and instead used a variable all_patches was because for the initial step I am extracting alternate patches. But later I am extracting all patches for the raw PatchMatch step. I understand of course that is resulting in unnecessary repetition. Will try to fix that later.